### ISSUE TITLE
/v1/me/player - context is missing most of the time

### ISSUE THREAD

**Issue by michaelherger** [spotify/web-api#602]
Issue found on July 6th 2017.

#### Endpoint(s):
* `GET /v1/me/player`

#### Scope(s):
* user-read-private
* user-follow-modify
* user-follow-read
* user-library-read
* user-library-modify
* user-top-read
* user-read-recently-played
* user-read-playback-state
* playlist-read-private
* playlist-read-collaborative
* playlist-modify-public
* playlist-modify-private

#### Steps to reproduce:
1. Call the me/player or me/player/currently-playing endpoint 
2. context is null unless a named playlist was playing. No context for playing an album, artist, or mix

#### Expected behaviour:
I'd expect a context for at least those items which are available through the API (album, artist tracks etc.)

#### Actual behaviour:
context: null

**Comment by mavvari** [comment]
Hi,

Can you clarify few questions about the following endpoint ?
"
End Point: /v1/me/player

Documentation: 
https://developer.spotify.com/web-api/get-information-about-the-users-current-playback/

The information returned is for the last known state, which means an inactive device could be returned if it was the last one to execute playback.
"

#1) I should be able to retrieve the last playback information, including 'context' and 'device' even if I am not playing currently. Correct ?

#2) Sometime I do see 'context' and 'device' fields as NULL, though I do see Track information all the time. On what circumstances those two fields could be NULL ? Could you please explain ?

#3) Would it be possible to add 'track_index' in 'Context' object ? 

Thanks,
- Murthy

**Comment by jscholes** [comment]
> I should be able to retrieve the last playback information, including 'context' and 'device' even if I am not playing currently. Correct ?

That's correct.  Assuming that your Spotify application or device has appropriately updated the playback state on the server, and that the updated state has propagated to whichever internal Spotify systems are responsible for making it available via the Web API.  It's perhaps worth noting that, from time to time, this doesn't seem to work as expected.  For example, the Spotify app on my iPhone and the one running on my desktop will often get out of sync when transferring playback, particularly when the desktop client has been running for more than a week.  Note that I don't work for Spotify or represent them in any way and this is entirely anecdotal.

> Sometime I do see 'context' and 'device' fields as NULL, though I do see Track information all the time. On what circumstances those two fields could be NULL ? Could you please explain?

The context will be null if the type of the currently playing context isn't supported by the Web API.  The Web API supports album, artist and playlist contexts, which means the context field won't be populated if you're playing a radio station, podcast, video or local file.  It's also possible that in some circumstances there won't actually be a context, for example if you've played a single track from an older or unofficial Spotify Connect device which doesn't support contexts.

Presumably the device field can be null if the device you're playing from isn't controllable through the Web API, e.g. a third party speaker or a television.

**Comment by elijaholmos** [comment]
I have the same issue. I made a call to /v1/me/player while I was playing an album (not in a private session), and it consistently returned a null context. The documentation states that context should be returned if the user's listening type is a "album, artist, or playlist" - but this seems to be only true for playlists. 

Has anyone else had a similar problem?
